src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ProvideCorrectArgumentsToFormattingMethods.cs

```diff
@@ -430,6 +448,27 @@ private static int FindParameterIndexOfName(ImmutableArray<IParameterSymbol> par
                 return -1;
             }
 
+            private int FindParameterIndexOfCompositeFormatStringSyntaxAttribute(ImmutableArray<IParameterSymbol> parameters)
+            {
+                for (var i = 0; i < parameters.Length; i++)
+                {
+                    ImmutableArray<AttributeData> attributes = parameters[i].GetAttributes();
+                    for (int j = 0; j < attributes.Length; j++)
+                    {
+                        if (StringSyntaxAttribute!.Equals(attributes[j].AttributeClass))
+                        {
+                            ImmutableArray<TypedConstant> arguments = attributes[j].ConstructorArguments;
+                            if (arguments.Length == 1 && CompositeFormat.Equals(arguments[0].Value))
+                            {
+                                return i;
+                            }
+                        }
+                    }
```

Comment:
NIT: if the parameter type is not string do not need to check the attributes
```suggestion
                    if (String.Equals(parameters[i].Type, SymbolEqualityComparer.Default))
                    {
                        foreach (AttributeData attribute in parameters[i].GetAttributes())
                        {
                            if (StringSyntaxAttribute!.Equals(attribute.AttributeClass, SymbolEqualityComparer.Default))
                            {
                                ImmutableArray<TypedConstant> arguments = attribute.ConstructorArguments;
                                if (arguments.Length == 1 && CompositeFormat.Equals(arguments[0].Value))
                                {
                                    return i;
                                }
                            }
                        }
                    }
```